src/pages/Table/Table.tsx

```diff
@@ -1,58 +1,64 @@
-import * as React from 'react';
+import Ract, { useState, useEffect } from 'react';
 import Table from '@mui/material/Table';
 import TableBody from '@mui/material/TableBody';
 import TableCell from '@mui/material/TableCell';
 import TableContainer from '@mui/material/TableContainer';
 import TableHead from '@mui/material/TableHead';
 import TableRow from '@mui/material/TableRow';
 import Paper from '@mui/material/Paper';
-import makeServer from '../../server';
-import { useState } from "react";
+import UserI from '../../types/User';
 
+interface ParamsI {
+  page: number;
+  limit: number;
+}
 
-//variables
-let page = -1;
-let limit = 10;
-let rows:any = [];
-
-//make server
-makeServer();
-//create data function
-function createData(
-  firstName: string,
-  lastName: string,
-  dob: string,
-  isAdmin: boolean,
-) {
-  return { firstName, lastName, dob, isAdmin,};
+interface DataI {
+  users: UserI[];
+  totalEntries: number;
 }
 
 export default function BasicTable() {
-  const [data, setData] = useState([])
-  async function getapi(url:string) {
-    // Storing response
-    const response = await fetch(url); 
-    // Storing data in form of JSON
-    let data = await response.json();
-    //pushing data in order
-    data.users.forEach((user: { firstName: string; lastName: string; dateOfBirth: string; isAdmin: boolean; }) => {
-      rows.push(createData(user.firstName, user.lastName, user.dateOfBirth, user.isAdmin));
-    });
-    setData(rows);
-    debugger;
-  }
-  //function for getting table filled for the irst time
-  const startFill =()=>{
-    if(page === -1){
-      page=0;
-      getapi('api/users/'+page+'/'+limit)
+  const initialDataState: DataI = {
+    users: [],
+    totalEntries: 0,
+  };
+  const [data, setData] = useState<DataI>(initialDataState);
+  const [params, setParams] = useState<ParamsI>({
+    page: 1,
+    limit: 10,
+  });
+
+  useEffect(() => {
+    async function getUsers(url: string, params?: ParamsI) {
+      const response = await fetch(url + params?.page + '/' + params?.limit);
+      const { users, totalEntries } = await response.json();
+
+      setData({ users: users.models, totalEntries: totalEntries });
+    }
+
+    getUsers('api/users/', params);
+  }, [params]);
+
+  const handlePreviousPage = () => {
+    if (params.page > 0) {
+      setParams((prevState) => ({ ...prevState, page: prevState.page - 1 }));
```

Comment:
We take previous state of our params object and spread it, then overwrite page with new value based on the one we remember from just while ago (prevState.page)